docs/DeveloperGuide.md

```diff
@@ -1,34 +1,514 @@
+
+![CircuitModelClass](diagrams/CircuitModelClassDiagram.png)
+
+<small><i>Figure 7</i></small>
+
+### `TemplateCircuitCommand`  <a name="temp-circ"></a>
+Given below is the sequence diagram for interactions within the `logic` and `template` components for the 
+`parse(template rl)` API call that implements the `template` command to create templates.
+
+![TemplateSequence](diagrams/TemplateSequenceDiagram.png)
+
+<small><i>Figure 8</i></small>
+
+### `SetCircuitCommand` <a name="set-circ"></a>
+
+Given below is the Sequence Diagram for interactions within the `Logic` component for the `parse("set r 500")` API call which implements the `set` command to set values of components.
+
+![SetSequence](diagrams/SetSequenceDiagram.png)
+
+<small><i>Figure 9</i></small>  
+
+### `AddCircuitCommand` <a name="add-circ"></a>
+
+Given below is the detailed Sequence diagram for interactions within the `logic`, `template` and `component` components 
+for the `parse("add parallel c 20")` API call that implements the `add` command to add components.
+
+![AddSequence](diagrams/AddSequenceDiagram.png)
+
+<small><i>Figure 10</i></small>  
+
+### `CalculateCircuitCommand` <a name="calc-circ"></a>
+The calculate command can be split into two distinct sequence diagrams. Given below is the sequence diagram for the
+calculate command that does not show the access of the `component` component and shows the interactions for the
+`parse("calc power")` API call that implements this version of the `calculate` command to calculate power.  
+
+![CalcPowerSequence](diagrams/CalcPowerSequenceDiagram.png)
+
+<small><i>Figure 11</i></small>  
+
+The second sequence diagram given below shows the detailed interaction that acesses the `component` class through the
+`parse("calc reff")` API call that implements this version of the `calculate` command to calculate effective resistance.
+
+![CalcReffSequence](diagrams/CalcReffSequenceDiagram.png)
+
+<small><i>Figure 12</i></small>  
+
+--------------------------------------------------------------------------------------------------------------------
+
+## Implementation of Boolean Commands (Praveen) <a name="bool-comd"></a>
+
+![BooleanCommandClass](diagrams/BooleanCommandClassDiagram.png)
+
+<small><i>Figure 13</i></small>
+
+The second major feature of the application is the implementation of Boolean logic commands, of which various noteworthy implementation details are explained in this section. The above diagram demonstrates the relationships between the various `Command` objects. Notice the similarities to the [Circuit Commands in the earlier section](#implementation-of-circuit-commands).
+
+There are six different logic gates that can be instantiated in the program:
+- `or, and, nor, nand, xor, xnor`
```

Comment:
Looking good so far! Might be better if the bullet point is omitted here, as it is only used once and the text inside it is already highlighted